js/src/UXCapture.js

```diff
@@ -0,0 +1,78 @@
+import ExpectedMark from "./ExpectedMark";
+import View from "./View";
+
+export default class UXCapture {
+  static attachTo(window) {
+    // allow running in Node.js environment
+    if (typeof window === "undefined") {
+      throw new Error("Must provide a valid window object");
+    }
+
+    // already have UX Capture object defined, reuse it
+    if (typeof window.UX !== "undefined") {
+      return window.UX;
+    }
+
+    // assign singleton instance to window object
+    window.UX = new UXCapture(window);
+
+    return window.UX;
+  }
+
+  constructor() {
+    this.onMark = null;
+    this.onMeasure = null;
+  }
+
+  /**
+   * Sets expected marks and corresponding zones for current view
+   *
+   * @TODO re-evaluate if we should allow multiple executions of this method
+   */
+  expect(zoneConfigs) {
```

Comment:
Since this PR is a breaking change (`label`->`name`), I'll go ahead and recommend a name change here as well - instead of `expect`, which is a little vague, I wonder if this method would be easier to understand as `createView` or `initView`, which makes it more clear _what_ is being done and _when_ it should be called (i.e. as soon as a view is 'started')

Reply:
It makes sense, but I would delay this change to next major version when we have SPA support when we'll really need multiple views - in this change View is just simplification of the library logic rather than really functional element.

At that point we will need need new API for supplying expected marks for multiple views (see https://meetup.atlassian.net/browse/WP-829)

The question comes though if we should really do `label` -> `name` change now and increment major to 2.x.x at all then?

Reply:
I'll leave createView to the moment when we have SPA implementation to make sure we have a good API there rather than rushing it in now, but will keep 2.x increment with `label`->`name` renaming in place.